tests/unit/merge_configs_test.py

```diff
@@ -0,0 +1,40 @@
+import pytest
+from unittest.mock import patch
+from pydeepmerge import merge_configs
+from pydeepmerge.errors import (
+    FileTypeError,
+    ParserNotAvailableError
+)
+
+
+@pytest.mark.unit
+@patch('pydeepmerge.parsers.json.json_parser')
+def test_merge_configs(mock_json_parser):
```

Comment:
This is a topic that's always up for debate but I would have two tests, one without overlapping keys and one with overlapping keys. That way if the test fails it's really quick to see what and where things are going wrong, but since this is a relatively simple function it might be overkill so your call.

Reply:
No you make a good point, I was really just being lazy and not doing things appropriately. I should realistically also create dictionaries with a depth > 1.

Would probably be good to add tests for non-strict as well but as it stands it would require creating test config files which I haven't gotten around to doing

Reply:
Oh yea I totally forgot this was supposed to handle nested dictionaries, yea you definitely need a test for that because that's an extra layer of complexity that you want to be sure you can handle

Reply:
tbf this is more so to make sure that it handles the file parsing correctly, the testing for the functionality of the `deep_merge` function is done in other tests

Reply:
Oh ok that's fair, my lack of knowledge of this project is definitely showing here, I don't know what's tested elsewhere :sweat_smile:

Reply:
And that's totally fair! Sorry for putting you on the spot